Filter dashboard by network

mykcryptodev opened the pull request:
My dashboard is getting crowded with lots of deployments and with Goerli being deprecated soon, I have a feeling that I'll be scrolling through a lot of deprecated contracts.

I created this filter-by-network feature and I'm happy to get feedback on it :) 

https://www.loom.com/share/012ed260351645c781326cf9553a9b9e

jnsdls commented:
idea is right but the filter should be part of the table itself, see: https://tanstack.com/table/v8/docs/examples/react/filters for an example

mykcryptodev commented:
> idea is right but the filter should be part of the table itself, see: https://tanstack.com/table/v8/docs/examples/react/filters for an example

@jnsdls I updated the filter to work directly on the table. The filter now replaces the header in the table but I'm happy to take direction on the design if that doesn't vibe. I ended up implementing the table's native pagination and removed the code that  simulated pagination. I did this in order for the filter to work across the entire dataset even if that data is not yet on the current page.

Let me know what ya think :)

MananTank commented:
@mykcryptodev - Can you add proper types to fix the build error?

jnsdls commented:
@mykcryptodev here's the error: 
<img width="1206" alt="image" src="https://github.com/thirdweb-dev/dashboard/assets/8204858/674f9259-8f4a-485c-8ce0-1d4de321655e">

mykcryptodev commented:
> @mykcryptodev here's the error: <img alt="image" width="1206" src="https://user-images.githubusercontent.com/8204858/237791996-674f9259-8f4a-485c-8ce0-1d4de321655e.png">

@jnsdls I fixed the type errors for the build

mykcryptodev commented on components/contract-components/tables/deployed-contracts.tsx:
```diff
@@ -86,10 +86,26 @@ export const DeployedContracts: React.FC<DeployedContractsProps> = ({
   contractListQuery,
   limit = 10,
 }) => {
+  const [showMoreLimit, setShowMoreLimit] = useState(limit);
```
@MananTank I like what you did with the design!

I think this commit introduced a regression bug though. I changed around how Show More works because the filter needs to be able to filter all of your contracts (even the ones that are not on the current page). Going back to this method of Show More will only filter contracts on the current page. Is there a good reason to revert back to this method of Show More?

MananTank replied:
@mykcryptodev I've taken care of it - `chainIdsWithDeployments` is calculated with `contractListQuery.data` and not the sliced data

MananTank replied:
my bad - you are right. we can't have show more the way list is filtered

mykcryptodev replied:
@MananTank Im on it - I can open a PR -- logged issue #1204

MananTank replied:
I pushed a fix - https://github.com/thirdweb-dev/dashboard/pull/1205

MananTank commented:
LGTM, Merging it
Thanks @mykcryptodev for your contribution 🔥
